### PR TITLE
Attendance refactor

### PR DESCRIPTION
**Pull request by BenHSmith13**
This refactors and adds specs for the Attendance app

### CODE REVIEW

File: client/js/attendance/components/main/student_list.jsx [hunk 1]
```diff
@@ -1,156 +1,159 @@
-"use strict";
+import { connect } from 'react-redux';
+import React from 'react';
+import _ from 'lodash';
+import canvasRequest from '../../../libs/canvas/action';
+import { list_users_in_course_users } from '../../../libs/canvas/constants/courses';
```
Comment (BenHSmith13) [inline client/js/attendance/components/main/student_list.jsx]: This is auto generated, would need to refactor things in another app to fix.

**Comment by BenHSmith13** [comment]
Hound Passed, I am so happy I could cry

File: client/js/attendance/reducers/student.js [hunk 2]
```diff
@@ -1,39 +1,35 @@
...
-    student.name = current.name;
-    student.sortable_name = current.sortable_name;
-    students[current.id] = student;
-    return students;
-  }, {});
-}
-
 export const initialState = () => ({
-  all:{}
+  all: [],
```
Comment (jbasdf) [inline client/js/attendance/reducers/student.js]: This should be an object not an array.

File: client/js/attendance/reducers/student.spec.js [hunk 3]
```diff
@@ -1,41 +1,44 @@
...
+  ];
 
   it('updates all students', () => {
     const action = {
       type: list_users_in_course_users.type + DONE,
-      payload:students
+      payload: students,
     };
-    const result = studentReducer(initialState(), action);
+    const result = studentReducer({ all: [] }, action);
```
Comment (jbasdf) [inline client/js/attendance/reducers/student.spec.js]: change array to an object

File: client/styles/global_styles.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+// This file exists to compile all css - less and scss - into one file
```
Comment (jbasdf) [inline client/styles/global_styles.js]: This isn't used. Let's delete it.

**Comment by jbasdf** [comment]
LGTM

**Pull request merged**